src/lib/queryVersions.ts

```diff
@@ -37,79 +37,68 @@ async function queryVersions(packageMap: Index<VersionSpec>, options: Options =
+      return Promise.reject(
+        new Error(
+          `Unsupported target "${targetResult}" for ${packageManagerName}. Supported version targets are: ` +
+            packageManagerSupportedVersionTargets.join(', ') +
+            (!isGithubDependency ? ' and custom distribution tags, following "@" (example: @next)' : ''),
+        ),
+      )
+    }
+
+    try {
       versionNew = await getPackageVersion(name, version, {
         ...options,
         // upgrade prereleases to newer prereleases by default
+        distTag,
         pre: options.pre != null ? options.pre : isPre(version),
+        retry: options.retry ?? 2,
       })
-    } else {
-      // set the getPackageVersion function from options.target
-      // TODO: Remove "as GetVersion" and fix types
-      const getPackageVersion = packageManager[targetResult as keyof typeof packageManager] as GetVersion
-      if (!getPackageVersion) {
-        const packageManagerSupportedVersionTargets = supportedVersionTargets.filter(t => t in packageManager)
-        return Promise.reject(
-          new Error(
-            `Unsupported target "${targetResult}" for ${
-              options.packageManager || 'npm'
-            }. Supported version targets are: ${packageManagerSupportedVersionTargets.join(', ')}`,
-          ),
-        )
-      }
 
-      try {
-        versionNew = await getPackageVersion(name, version, {
-          ...options,
-          // upgrade prereleases to newer prereleases by default
-          pre: options.pre != null ? options.pre : isPre(version),
-          retry: options.retry ?? 2,
-        })
-        versionNew = npmAlias && versionNew ? createNpmAlias(name, versionNew) : versionNew
-      } catch (err: any) {
-        const errorMessage = err ? (err.message || err).toString() : ''
-        if (errorMessage.match(/E404|ENOTFOUND|404 Not Found/i)) {
-          return {
-            error: `${errorMessage.replace(
-              / - Not found$/i,
-              '',
-            )}. Either your internet connection is down or unstable and all ${
-              options.retry
-            } retry attempts failed, or the registry is not accessible, or the package does not exist.`,
-          }
-        } else {
-          // print a hint about the --timeout option for network timeout errors
-          if (!process.env.NCU_TESTS && /(Response|network) timeout/i.test(errorMessage)) {
-            console.error(
-              '\n\n' +
-                chalk.red(
-                  'FetchError: Request Timeout. npm-registry-fetch defaults to 30000 (30 seconds). Try setting the --timeout option (in milliseconds) to override this.',
-                ) +
-                '\n',
-            )
-          }
-
-          throw err
+      versionNew = !isGithubDependency && npmAlias && versionNew ? createNpmAlias(name, versionNew) : versionNew
+    } catch (err: any) {
```

Comment:
That whole part now applies to github too. Didn't change texts, but they are more-or-less general